CommandLine/testing/Makefile

```diff
@@ -4,11 +4,14 @@ TARGET := ../../test-runner
 TESTCASES_DIR := $(SRC_DIR)/Tests
 SHARED_SRC_DIR := ../../shared
 
-ifeq ($(OS), Linux)
-	OS_SRCS=$(wildcard ./Platform/*-X11.cpp)
-	OS_LIBS=-lX11 -lstdc++fs
-else ifeq ($(OS), Darwin)
+ifeq ($(OS), Darwin)
 	OS_SRCS=./Platform/TestHarness-Cocoa.cpp
+else ifeq ($(UNIX_BASED), true)
+	OS_SRCS=$(wildcard ./Platform/*-X11.cpp)
+	OS_LIBS:=$(shell pkg-config x11 --libs)
+	ifeq ($(OS), Linux)
```

Comment:
@JoshDreamland considering FreeBSD uses all the same deps and API's as Linux, with the exception being the lack of real joystick/gamepad api, it should work I'd imagine. Though I'd like to know what I would need to do in order to test that theory, if it can be tested from the command line/emake or whatever. Never done it before.

Reply:
Clang doesn't need the OS_LIBS+=-lstdc++fs iirc.